kolibri/core/discovery/utils/network/client.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+import requests
+from six.moves.urllib.parse import urljoin
+
+from . import errors
+from .urls import get_normalized_url_variations
+
+logger = logging.getLogger(__name__)
+
+
+class NetworkClient(object):
+
+    def __init__(self, base_url=None, address=None, **kwargs):
+        """If an explicit base_url is already known, provide that. If a vague address is provided, we can try to infer the base_url"""
+        if not base_url and not address:
+            raise Exception("You must provide either a `base_url` or `address` argument")
+        self.session = requests.Session(**kwargs)
+        if base_url:
+            self.base_url = self._attempt_connections([base_url])
+        else:
+            # normalize the URL and try a number of variations until we find one that's able to connect
+            logger.info("Attempting connections to variations of the URL: {}".format(base_url))
```

Comment:
will we actually log the `base_url` if it not supposed to be defined by the time we get here?